src/main/java/frc/robot/commands/rests/restUtils/RESTHandler.java

```diff
@@ -0,0 +1,272 @@
+package frc.robot.commands.rests.restUtils;
+
+import edu.wpi.first.util.datalog.DataLog;
+import edu.wpi.first.util.datalog.StringLogEntry;
+import edu.wpi.first.util.sendable.Sendable;
+import edu.wpi.first.util.sendable.SendableBuilder;
+import edu.wpi.first.util.sendable.SendableRegistry;
+import edu.wpi.first.wpilibj.DataLogManager;
+import edu.wpi.first.wpilibj.Timer;
+import edu.wpi.first.wpilibj2.command.Subsystem;
+import frc.robot.commands.rests.RESTCommand;
+import frc.robot.commands.rests.restAnnotations.*;
+
+import javax.inject.Inject;
+import java.lang.annotation.AnnotationTypeMismatchException;
+import java.lang.reflect.Field;
+import java.lang.reflect.Method;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Objects;
+
+public class RESTHandler implements Sendable, AutoCloseable {
+    private final List<Object> rests;
+    private final HashMap<Object, List<Method>> restTests;
+
+    private Object currentREST;
+    private List<Method> currentTestList;
+    private Method currentTest;
+    private List<Subsystem> currentRequirements;
+    private static Runnable init = () -> {
+    };
+    private static Runnable execute = () -> {
+    };
+    private static Runnable end = () -> {
+    };
+
+    private final String[][] results;
+
+    private DataLog log;
+    private StringLogEntry resultLog;
+    private static Timer timer;
+
+    private boolean newREST = true;
+    private static boolean testFinished = true;
+    private boolean newTest = true;
+    private int restIndex = 0;
+    private int testIndex = 0;
+
+    @Inject
+    public RESTHandler(List<Object> rests) {
+        this.rests = rests;
+        restTests = new HashMap<>();
+        results = new String[rests.size()][];
+
+        int i = 0;
+        for (Object rest : rests) {
+            if (Objects.isNull(rest)) {
+                throw new NullPointerException("SubsystemTest for SubsystemTestHandler is null");
+            }
+
+            Class<?> restClass = rest.getClass();
+            if (!restClass.isAnnotationPresent(RobotEnabledSelfTest.class)) {
+                throw new AnnotationTypeMismatchException(null, "This class is not annotated as a SubsystemTest");
+            }
+
+            restTests.put(rest, new ArrayList<>());
+
+            for (Method method : restClass.getDeclaredMethods()) {
+                if (method.isAnnotationPresent(Test.class)) {
+                    method.setAccessible(true);
+                    restTests.get(rest).add(method);
```

Comment:
Performance is probably not as big of a deal here, but it would be faster to keep a reference to the array list and use that reference directly instead of retrieving it from the map each time.